internal/kinds/capp/controllers/controller.go

```diff
@@ -127,6 +135,55 @@ func (r *CappReconciler) findCappFromHostname(ctx context.Context, object client
 	return []reconcile.Request{request}
 }
 
+// findAllCapps maps reconciliation requests to all Capp instances in the system.
+func (r *CappReconciler) findAllCapps(ctx context.Context, object client.Object) []reconcile.Request {
+	var requests []reconcile.Request
+
+	var cappList cappv1alpha1.CappList
+	if err := r.Client.List(ctx, &cappList, &client.ListOptions{}); err != nil {
+		log := log.FromContext(ctx)
+		log.Error(err, "Failed to list Capp resources")
+		return nil
+	}
+
+	for _, capp := range cappList.Items {
+		requests = append(requests, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: capp.Namespace,
+				Name:      capp.Name,
+			},
+		})
+	}
+
+	log.FromContext(ctx).Info("Found Capp resources", "count", len(cappList.Items), "requests", requests)
+	return requests
+}
+
+// DNSConfigChangedPredicate Defines a predicate that triggers the reconcile only when dnsConfig is changed
+func DNSConfigChangedPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldCappConfig, ok := e.ObjectOld.(*cappv1alpha1.CappConfig)
+			if !ok {
+				log.Log.Info("old object is not a CappConfig")
+				return false
+			}
+
+			newCappConfig, ok := e.ObjectNew.(*cappv1alpha1.CappConfig)
+			if !ok {
+				log.Log.Info("new object is not a CappConfig")
+				return false
+			}
+
+			if oldCappConfig.Spec.DNSConfig != newCappConfig.Spec.DNSConfig {
```

Comment:
Does this work, I might be wrong? Also logging here is unnecessary.

Reply:
Yes it works, i checked it last week